pkg/controller/infrastructure/infraflow/reconcile.go

```diff
@@ -199,13 +199,17 @@ func (fctx *FlowContext) ensureNetwork(ctx context.Context) error {
 }
 
 func (fctx *FlowContext) ensureConfiguredNetwork(_ context.Context) error {
-	network, err := fctx.access.GetNetworkByID(*fctx.config.Networks.ID)
+	networkId := *fctx.config.Networks.ID
+	network, err := fctx.access.GetNetworkByID(networkId)
 	if err != nil {
 		fctx.state.Set(IdentifierNetwork, "")
 		fctx.state.Set(NameNetwork, "")
 		return err
 	}
-	fctx.state.Set(IdentifierNetwork, *fctx.config.Networks.ID)
+	if network == nil {
+		return fmt.Errorf("network with ID '%s' was not found", networkId)
```

Comment:
Returning error is okay, but we should make sure that we give an ERR_INFRA_DEPENDENCIES error